### DISCUSSION TITLE
kumascript tests and cleanup

### DISCUSSION THREAD

**Discussion by schalkneethling** [mdn/mdn-community#609] [Content]
So, turns out the kumascript tests are a blocker for our move to ESM. I spoke with a few people and it seems we can get around this by dropping the existing unit tests, and instead, testing the macros using headless tests. The thinking is that what we really want to test is that the output of a macro is what is expected. I am wondering whether we can:

1. Use the Kitchen Sink page for this purpose or,
2. Should we create individual fixture pages for the various macros and run the tests against each of those?

Are there any macros we know we can drop or render out?
Also, the question of deprecating macros has been brought up. We can do this and show the message for folks running in non-production mode.

Would love to get people feedback on the above. Thank you.

**Comment by pepelsbey** [comment]
Closing as outdated/irrelevant.
